Add support for using ansible 2.7/python 2/centos7 as controller (#87)
linux-system-roles/test-harness: README.md

To make changes to the CI environment, the changes should be first committed
to the `master` branch, and deployed to the `staging` environment.  Once they
are confirmed to be working, the changes should be merged to the `production`

> branch and deployed to the `production` environment.  For the centos7 based tests, use `linux-system-roles-centos7` for production and

Reviewer: The previous lines seem to be wrapped but the new one is not.

Author: https://github.com/linux-system-roles/test-harness/pull/87/commits/a1f7e36ec876eb927bb6ddcd25ba619b13dc88c0

will be executed using the Ansible `raw` module (so the two examples above are
exactly equivalent).

Use the `min_ansible_version` key to indicate the minimum version of

> ansible that can be used.  For example, if you cannot use ansible 2.7

Reviewer: ```suggestion
ansible that can be used.  For example, if you need to use at least ansible 2.7
```

Author: https://github.com/linux-system-roles/test-harness/pull/87/commits/a1f7e36ec876eb927bb6ddcd25ba619b13dc88c0

to manage Fedora 32 hosts, use something like this:
```json
    {
      "name": "fedora-32",
      "source": "http://download.fedoraproject.org/pub/fedora/linux/releases/32/Cloud/x86_64/images/Fedora-Cloud-Base-32-1.6.x86_64.qcow2",

>       "upload_results": false,

Reviewer: This does not need to be false, does it? Also `false` should be the default, so this seems to be better:
```suggestion
      "upload_results": true,
```

Author: https://github.com/linux-system-roles/test-harness/pull/87/commits/a1f7e36ec876eb927bb6ddcd25ba619b13dc88c0

      "min_ansible_version": "2.8"
    },
```

> This means the CI environment will not run tests using ansible 2.7 to

Reviewer: ```suggestion
This means the CI environment will not run tests using ansible 2.7 or earlier to
```

Author: fixed in https://github.com/linux-system-roles/test-harness/pull/87/commits/e08a9eff0dd3e7052bcbad465dfdc921caf1464f

...
$ oc exec linux-system-roles-12-ddtrs -- cat /var/log/test-harness_linux-system-roles-12-ddtrs.log
```

## run-tests Environment Variables

The easiest way to customize the pod is via environment variables.  The
environment variables correspond to the command line arguments to `run-tests`.
The environment variables will override the command line parameters, if both
are used.  In general, the name of the environment variable is the name of the
command line argument, in all upper case, with `_` instead of `-`, and with
the prefix `TEST_HARNESS_`.  For example, the command line option
`--use-images` can be set via the env. var. `TEST_HARNESS_USE_IMAGES`.  See
`run-tests` usage for the complete list.  In addition, there are some other
env. vars which are useful for debugging:
* `TEST_HARNESS_CONFIG_NAME` - replaces `config["name"]` - used for the name in
the PR status message
* `TEST_HARNESS_REPOSITORIES` - comma delimited list of repos to override
`config["repositories"]`
* `TEST_HARNESS_KEEP_RESULTS` - if `true`, this will cause the test not to
delete the temp directory where it holds the results before uploading them

> For example, in a running OpenShift cluster, if you wanted to use a custom name for the PR status:

Reviewer: The lines seem to be wrapped inconsistently.

Author: https://github.com/linux-system-roles/test-harness/pull/87/commits/a1f7e36ec876eb927bb6ddcd25ba619b13dc88c0

```
oc set env dc/linux-system-roles-staging TEST_HARNESS_CONFIG_NAME=my_custom_name
```

> This will cause the staging pods to be redeployed and use `my_custom_name` for the status reporting.  To reset:

Reviewer: The lines seem to be wrapped inconsistently.

Author: https://github.com/linux-system-roles/test-harness/pull/87/commits/a1f7e36ec876eb927bb6ddcd25ba619b13dc88c0
